From d58c593183a8606e1caa168c430d43ec81266a64 Mon Sep 17 00:00:00 2001
From: Xiaokang Qin <xiaokang.qin@intel.com>
Date: Wed, 14 Nov 2012 11:21:33 +0800
Subject: [PATCH 03/22] [PORT FROM MAIN] clean droidboot & recovery platform
 related codes

BZ: 46856

Move all droidboot rules into other makefile. For recovery and ota
stuff, introduce conditional compilation to differenciate intel
specific code (put in other makefile) from google code. New
makefiles are located in vendor/intel/build/tasks

[BOOT CAMP FEATURE: PUPDR]

Change-Id: Iaf9a00277a77160fc73c313047488cd76eb21b09
Reviewed-on: http://android.intel.com:8080/74478
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 core/Makefile |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 3117c3f..be1330c 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -578,6 +578,7 @@ endef
 # Recovery image
 
 # If neither TARGET_NO_KERNEL nor TARGET_NO_RECOVERY are true
+ifneq ($(TARGET_MAKE_NO_DEFAULT_RECOVERY),true)
 ifeq (,$(filter true, $(TARGET_NO_KERNEL) $(TARGET_NO_RECOVERY) $(BUILD_TINY_ANDROID)))
 
 INSTALLED_RECOVERYIMAGE_TARGET := $(PRODUCT_OUT)/recovery.img
@@ -682,6 +683,7 @@ endif
 
 .PHONY: recoveryimage
 recoveryimage: $(INSTALLED_RECOVERYIMAGE_TARGET) $(RECOVERY_RESOURCE_ZIP)
+endif # TARGET_MAKE_NO_DEFAULT_RECOVERY
 
 ifneq ($(BOARD_NAND_PAGE_SIZE),)
 mkyaffs2_extra_flags := -c $(BOARD_NAND_PAGE_SIZE)
@@ -1022,6 +1024,8 @@ otatools: $(OTATOOLS)
 # This zip can be used to create an OTA package or filesystem image
 # as a post-build step.
 #
+ifneq ($(TARGET_MAKE_NO_DEFAULT_OTA_PACKAGE),true)
+
 name := $(TARGET_PRODUCT)
 ifeq ($(TARGET_BUILD_TYPE),debug)
   name := $(name)_debug
@@ -1225,6 +1229,7 @@ endif    # recovery_fstab is defined
 endif    # TARGET_NO_KERNEL != true
 endif    # TARGET_DEVICE != generic*
 endif    # TARGET_PRODUCT != sdk
+endif # TARGET_MAKE_NO_DEFAULT_OTA_PACKAGE
 
 # -----------------------------------------------------------------
 # A zip of the tests that are built when running "make tests".
-- 
1.7.9

